### PR TITLE
[mlir] Lower math dialect later in gpu-lower-to-nvvm-pipeline

### PR DESCRIPTION
**Pull request by grypp**
This PR moves lowering of math dialect later in the pipeline. Because math dialect is lowered correctly by `createConvertGpuOpsToNVVMOps` for GPU target, and it needs to run it first.

### CODE REVIEW

File: stmatrix/everything-casted.mlir [hunk 1]
```diff
@@ -0,0 +1,1267 @@
+// -----// IR Dump After ConvertNVGPUToNVVMPass (convert-nvgpu-to-nvvm) //----- //
```
Comment (apaszke) [inline stmatrix/everything-casted.mlir:1]: I assume this is an accidental inclusion to this PR?

File: mlir/test/Dialect/GPU/test-nvvm-pipeline.mlir [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+// CHECK-LABEL: llvm.func @test_math(%arg0: f32) {
+func.func @test_math(%arg0 : f32) {
+    %c2 = arith.constant 2 : index
+    %c1 = arith.constant 1 : index
+    // CHECK: gpu.launch_func  @test_math_kernel::@test_math_kernel
+    // CHECK: gpu.binary @test_math_kernel  [#gpu.object<#nvvm.target
+    gpu.launch 
+        blocks(%0, %1, %2) in (%3 = %c1, %4 = %c1, %5 = %c1) 
+        threads(%6, %7, %8) in (%9 = %c2, %10 = %c1, %11 = %c1) { 
+         %s1 = math.exp %arg0 : f32
```
Comment (apaszke) [inline mlir/test/Dialect/GPU/test-nvvm-pipeline.mlir:12]: Add a check that there is a call to the libdevice function?

File: mlir/test/Dialect/GPU/test-nvvm-pipeline.mlir [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// RUN: mlir-opt -gpu-lower-to-nvvm-pipeline="cubin-format=isa" -split-input-file %s | FileCheck %s
+
+// CHECK-LABEL: llvm.func @test_math(%arg0: f32) {
+func.func @test_math(%arg0 : f32) {
+    %c2 = arith.constant 2 : index
+    %c1 = arith.constant 1 : index
```
Comment (apaszke) [inline mlir/test/Dialect/GPU/test-nvvm-pipeline.mlir:6]: Perhaps add an exp here as well and check that it lowers to llvm intrinsics?

**Review by apaszke — APPROVED** [review]

**Comment by grypp** [comment]
Buildbot did not like the way I check PTX. I'll fix this quickly.

**Pull request merged**